devserver/__init__.py

```diff
@@ -0,0 +1,361 @@
+        participant_count=1
+    ),
+    make_event(
+        created_at=time_from_english('30 minutes ago'),
+        start_time=time_from_english('1 day from now'),
+        end_time=time_from_english('1 day, 1 hour from now'),
+        created_by=random.choice(users),
+        participant_count=3
+    ),
+]
+
+api_data = deepcopy(original_api_data)
+
+@app.template_filter('dtformat')
+def dtformat(value, format='%-m-%d-%Y at %H:%M'):
+    return dateutil.parser.parse(value).strftime(format)
+
+
+def sort_by_start_time(events):
+    return sorted(events, key=lambda e: dateutil.parser.parse(e['start_time']))
+
+@app.route('/')
+def index():
+    return render_template('index.html', title="Index", events=sort_by_start_time(api_data), flask_debug_mode=os.getenv('FLASK_DEBUG'))
+
+@app.route('/reset', methods=['POST'])
+def reset():
+    global next_event_id, api_data
+    next_event_id = 1
+    api_data = deepcopy(original_api_data)
+
+    return redirect(url_for('index'))
+
+@app.route('/create', methods=['POST'])
+def create():
+    days = random.randint(1, 10)
+    hours = random.randint(1, 10)
+
+    start_time = time_from_english("{} days, {} hours from now".format(days, hours))
+    end_time = start_time + datetime.timedelta(minutes=30)
+
+    event = make_event(
+        created_at=utcnow(),
+        start_time=start_time,
+        end_time=end_time,
+        created_by=random.choice(users),
+        participant_count=random.randint(0, len(users) - 1)
+    )
+
+    api_data.append(event)
+
+    return redirect(url_for('index'))
+
+
+# parses a date and sets tzinfo to UTC if it is not set.
+# otherwise we'll fail later
+def parse_date(s):
+    date = dateutil.parser.parse(s)
+
+    if not date.tzinfo:
+        date = date.replace(tzinfo=pytz.utc)
+
+    return date
+
+def created_on_or_after(event, date):
+    return dateutil.parser.parse(event['created_at']) >= date
+
+def find_event(id):
+    return next((e for e in api_data if e['id'] == id), None)
+
+@app.route('/api/v1/events')
+def events():
+    if 'created_at_or_after' in request.args:
+        date = parse_date(request.args['created_at_or_after'])
+        events = [event for event in api_data if created_on_or_after(event, date)]
+        return jsonify(sort_by_start_time(events))
+    elif 'ids' in request.args:
+        ids = json.loads(request.args['ids'])
+        return jsonify([find_event(id) for id in ids if find_event(id) is not None])
+    else:
+        return jsonify([])
+
+def not_found():
+    response = jsonify({"message":"not_found"})
+    response.status_code = 404
+    return response
+
+def user_not_specified():
+    response = jsonify({"message": "user_not_specified"})
+    response.status_code = 422
+    return response
+
+@app.route('/api/v1/events/<int:id>', methods=['GET'])
+def event(id):
+    event = find_event(id)
+    if event == None:
+        return not_found()
+
+    include_participants = request.args.get('include_participants', '').lower()
```

Comment:
Hmm… one thing we could do is have a `render_event` and `render_event` method that wraps the `jsonify`, so you'd do `return render_event(event)` and it would be defined as:

```py
def filter_participants(event):
    include_participants = request.args.get('include_participants', '').lower()
    if include_participants == 'true':
        return event
    else:
        return {k: v for k, v in event.items() if k != 'participants'}

def render_event(event):
    return jsonify(filter_participants(event))

def render_events(events):
    return jsonify([filter_participants(event) for event in events])
```